addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -26,6 +26,9 @@ const GalleryWidget = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onClickImg: function (ev) {
+        if (document.querySelector('.modal[aria-labbelledby="Image Gallery Dialog"]')) {
```

Comment:
The `aria-labbelledby="Image Gallery Dialog"` which is in the XML template makes no sense:
1. `aria-labelledby` is meant to reference a DOM element, not contain a random string. Here what is meant to be used is `aria-label`.
2. `aria-labelledby` has not even been written correctly... there should not be 2 "b" (funny enough, "labelledby" actually is a typo too (should be "labeledby" in English but it was kept as part of the HTML specs :grimacing:)

Targeting this wrong attribute in JS is wrong for multiple reasons:
1. You apparently don't know about this attribute's purpose, otherwise you would have caught that it was wrongly written in the first place -> don't write stuff you don't understand. Search the doc about it and then you can make an educated fix about the lines you touch.
2. Even if this was using the right attribute (`aria-label`), it would be translated: thus targeting it in JS would only work in English :grimacing: 
3. This code would have a very huge chance of becoming broken or dead code if we ever review the XML (like to fix the XML use of the random `aria-labbelledby` attribute): most people won't think about grepping that in the codebase to find your line of code -> try to code more defensively, especially when you have no plan in covering that bug fix with a test.

In the end, why do you need to search the DOM to check if a modal is already opened? This code is apparently in charge of creating that modal, can't you keep a reference to it or have a boolean property to know at all times if you have a modal opened?

You can also make a PR for master about the aria-label stuff :+1:

Reply:
Ok, thanks for the explanation :+1: . It should be correct now and here is the PR to correct the XML on master: https://github.com/odoo/odoo/pull/98401